src/model.cpp

```diff
@@ -1928,10 +1967,38 @@ void Model::fdJrzdsigma(const int ie, const int nroots, const realtype t,
     }
 }
 
+void Model::fspl(const realtype t) {
+    for (int ispl = 0; ispl < nspl; ispl++) {
+        if (splines_[ispl].get_logarithmic_parametrization()) {
+            state_.spl_[ispl] = std::exp(splines_[ispl].getValue(t));
+        } else {
+            state_.spl_[ispl] = splines_[ispl].getValue(t);
+        }
+    }
+}
+
+void Model::fsspl(const realtype t) {
+    derived_state_.sspl_.zero();
+    realtype *sspl_data = derived_state_.sspl_.data();
+    for (int ip = 0; ip < nplist(); ip++) {
+        for (int ispl = 0; ispl < nspl; ispl++) {
+            if (splines_[ispl].get_logarithmic_parametrization()) {
+                sspl_data[ispl + nspl * ip] =
+                    state_.spl_[ispl] * splines_[ispl].getSensitivity(t, ip);
+            } else {
+                sspl_data[ispl + nspl * ip] =
+                    splines_[ispl].getSensitivity(t, ip);
+            }
```

Comment:
Can/should that `if` not go into `getSensitivity`?

Reply:
@lcontento ?

Reply:
If we put the `if` inside then we would have to recompute the spline value for each parameter, while in this way we can recycle it.

Reply:
As mentioned in the other comment, I honestly think the `if` belongs logically inside the spline methods, but with the current method signature I would not move it. If we had a method that computes both spline value and spline sensitivities at the same time it would be better though in my opinion.

Reply:
see comment regarding same problem in sensitivity computation

Reply:
If I understand it correctly, this could be addressed by having `get_{value,sensitivity}` and `get_unscaled_{value,sensitivity}`? Here, we'd use the former, and where required inside `splinesfunctions.cpp`, we'd use the latter?